crates/services/producer/src/block_producer.rs

```diff
@@ -404,7 +409,17 @@ where
             if total_cost > gas_limit {
                 break;
             } else {
-                new_best = DaBlockHeight(height);
+                let transactions_number = self
+                    .relayer
+                    .get_transactions_number_for_block(&DaBlockHeight(height))
+                    .await?;
+                total_transactions =
+                    total_transactions.saturating_add(transactions_number);
+                if total_transactions > transactions_limit {
+                    break;
+                } else {
+                    new_best = DaBlockHeight(height);
+                }
             }
```

Comment:
It seems like we can simplify it to be 

```
if gas_check {
    break;
}

if tx_number_check {
    break;
}

new_best = DaBlockHeight(height);
```

Reply:
Done, simplified it a bit more in 6b5008d8aa2b755c670958f1de70cd854a7b7a2c